### ISSUE TITLE
Add support for custom initialization in Enum class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7029]
Enum classes support custom initialization. 

```python
from enum import Enum


class Planet(Enum):
    def __init__(self, value: int, mass: float, radius: float):
        self._value_ = value
        self.mass = mass
        self.radius = radius

    MERCURY = (1, 3.303e23, 2.4397e6)
    VENUS = (2, 4.869e24, 6.0518e6)
    EARTH = (3, 5.976e24, 6.37814e6)
    MARS = (6.421e23, 3.3972e6)  # Should be type checker error
    JUPITER = 5  # Should be type checker error

reveal_type(
    Planet.MERCURY.value
)  # Revealed type should be Literal[1] or int
```

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
